src/platform_impl/web/web_sys/canvas.rs

```diff
@@ -465,7 +468,34 @@ impl Common {
     }
 
     pub fn request_fullscreen(&self) {
-        *self.wants_fullscreen.borrow_mut() = true;
+        #[wasm_bindgen]
+        extern "C" {
+            type ElementExt;
+
+            #[wasm_bindgen(catch, method, js_class = "Element", js_name = requestFullscreen)]
+            fn request_fullscreen(this: &ElementExt) -> Result<JsValue, JsValue>;
+
+            type PromiseExt;
+
+            #[wasm_bindgen(method)]
+            fn catch(this: &PromiseExt, cb: &JsValue) -> Promise;
+        }
+
+        let raw: &ElementExt = self.raw.unchecked_ref();
+
+        // This should return a `Promise`, but some Safari <v16.4 is not up-to-date with the spec.
+        match raw.request_fullscreen() {
+            Ok(value) if !value.is_undefined() => {
+                let promise: PromiseExt = value.unchecked_into();
+                let wants_fullscreen = self.wants_fullscreen.clone();
+                let cb = Closure::once_into_js(move || *wants_fullscreen.borrow_mut() = true);
+                // We use a custom `catch` function here because we don't
+                // want to deal with an async cleanup of the `Closure`.
+                let _ = promise.catch(&cb);
```

Comment:
This workaround of dealing with the `Closure` could be removed if we ever pull in `wasm-bindgen-futures`.

Reply:
I used `wasm-bindgen-futures` instead.